lib/Post.php

```diff
@@ -734,224 +734,16 @@ public function field_object( $field_name ) {
-				$post_meta = empty( $field_name ) ? [] : null;
-			}
-		}
-
-		if ( $args['apply_filters'] ) {
-			/**
-			 * Filters the value for a post meta field.
-			 *
-			 * This filter is used by the ACF Integration.
-			 *
-			 * @example
-			 * ```php
-			 * add_filter( 'timber/post/meta', function( $post_meta, $post_id, $field_name, $post ) {
-			 *     if ( 'event' === $post->post_type ) {
-			 *         // Do something special.
-			 *         $post_meta['foo'] = $post_meta['foo'] . ' bar';
-			 *     }
-			 *
-			 *     return $post_meta;
-			 * }, 10, 4 );
-			 * ```
-			 *
-			 * @see   \Timber\Post::meta()
-			 * @since 2.0.0
-			 *
-			 * @param string       $post_meta  The field value.
-			 * @param int          $post_id    The post ID.
-			 * @param string       $field_name The name of the meta field to get the value for.
-			 * @param \Timber\Post $post       The post object.
-			 * @param array        $args       An array of arguments.
-			 */
-			$post_meta = apply_filters(
-				'timber/post/meta',
-				$post_meta,
-				$this->ID,
-				$field_name,
-				$this,
-				$args
-			);
-
-			/**
-			 * Filters the value for a post meta field.
-			 *
-			 * @deprecated 2.0.0, use `timber/post/meta`
-			 */
-			$post_meta = apply_filters_deprecated(
-				'timber_post_get_meta_field',
-				array( $post_meta, $this->ID, $field_name, $this ),
-				'2.0.0',
-				'timber/post/meta'
-			);
-
-			/**
-			 * Filters post meta data fetched from the database.
-			 *
-			 * @deprecated 2.0.0, use `timber/post/meta`
-			 */
-			$post_meta = apply_filters_deprecated(
-				'timber_post_get_meta',
-				array( $post_meta, $this->ID, $this ),
-				'2.0.0',
-				'timber/post/meta'
-			);
-
-			// Maybe convert values to Timber objects.
-			$post_meta = $this->convert($post_meta);
-		}
-
-		return $post_meta;
-	}
-
-	/**
-	 * Gets a post meta value directly from the database.
-	 *
-	 * Returns a raw meta value or all raw meta values saved in the post meta database table. In
-	 * comparison to `meta()`, this function will return raw values that are not filtered by third-
-	 * party plugins.
-	 *
-	 * Fetching raw values for all custom fields will not have a big performance impact, because
-	 * WordPress gets all meta values, when the first meta value is accessed.
-	 *
-	 * @api
-	 * @since 2.0.0
-	 *
-	 * @param string $field_name Optional. The field name for which you want to get the value. If
-	 *                           no field name is provided, this function will fetch values for all
-	 *                           custom fields. Default empty string.
-	 * @param array  $args       Optional. An array of args for `Post::meta()`. Default empty array.
-	 *
-	 * @return null|mixed The meta field value(s). Null if no value could be found, an empty array
-	 *                    if all fields were requested but no values could be found.
-	 */
-	public function raw_meta( $field_name = '', $args = array() ) {
-		return $this->meta( $field_name, array_merge(
```

Comment:
Could we leave in the `meta()` and `raw_meta()` functions and use calls to `parent::fetch_meta()` so we can profit from better documentation? Because otherwise I think we’re making it harder for users, e.g. if `meta()` and `raw_meta()` isn’t listed in [`Timber\Post`](https://timber.github.io/docs/v2/reference/timber-post/).

I’ve answered countless support questions for how to use meta correctly and I think we should be very clear on how to use the meta functions and where they are available. 

If they don’t appear in the documentation for the different objects, but only in a separate document for the `CoreEntity` class, I think we’re making it harder for developers to understand them.

And we couldn’t link to a specific object function like here: https://timber.github.io/docs/v2/reference/timber-post/#meta.

(Ideally, we’d use `@example` more in the description of all these objects, but that’s another todo.)

Reply:
I also thought about this case and I think that somehow I make this work in the documentation so that the `meta()` and `raw_meta()` methods still appear in the documentation under the various objects. I can probably import them as inherited method somehow. This shouldn’t hold this pull request back.